bird_classify.py

```diff
@@ -101,43 +103,74 @@ def main():
     'alarm' if a defined label is detected."""
     args = user_selections()
     print("Loading %s with %s labels."%(args.model, args.labels))
-    engine = ClassificationEngine(args.model)
-    labels = load_labels(args.labels)
+    interpreter = make_interpreter(args.model)
+    interpreter.allocate_tensors()
+    labels = read_label_file(args.labels)
+    output_tensors_size = interpreter.get_output_details()[0]['shape'][0]
+
+    if output_tensors_size != 1:
+        raise ValueError(
+            ('Classification model should have 1 output tensor only!'
+             'This model has {}.'.format(output_tensors_size)))
     storage_dir = args.storage
-
-    #Initialize logging file
+    # Initialize logging file
     logging.basicConfig(filename='%s/results.log'%storage_dir,
                         format='%(asctime)s-%(message)s',
                         level=logging.DEBUG)
-
     last_time = time.monotonic()
     last_results = [('label', 0)]
-    def user_callback(image,svg_canvas):
+
+    def user_callback(image, svg_canvas):
         nonlocal last_time
         nonlocal last_results
         start_time = time.monotonic()
-        results = engine.classify_with_image(image, threshold=args.threshold, top_k=args.top_k)
+        input_tensor_shape = interpreter.get_input_details()[0]['shape']
+        if (input_tensor_shape.size != 4 or
+                input_tensor_shape[0] != 1):
+            raise RuntimeError(
+                'Invalid input tensor shape! Expected: [1, height, width, channel]')
+        _, height, width, channel = input_tensor_shape
+        img = image.resize((width, height), Image.NEAREST)
+        # Handle color space conversion.
+        if channel == 1:
+            img = img.convert('L')
+        elif channel == 3:
+            img = img.convert('RGB')
+        else:
+            raise ValueError(
+                'Invalid input tensor channel! Expected: 1 or 3. Actual: %d' % channel)
+
+        input_tensor = np.asarray(img).flatten()
+        if args.top_k <= 0:
+            raise ValueError('top_k must be positive!')
+        run_inference(interpreter, input_tensor)
```

Comment:
Can you please try to use `set_resized_input` function from https://github.com/google-coral/pycoral/blob/master/pycoral/adapters/common.py here?